shub/bootstrap.py

```diff
@@ -0,0 +1,138 @@
+    "" % EXAMPLE_REPO)
+
+HELP = """
+Through custom images, Scrapinghub allows you to run crawlers written in any
+language you want. To get you started, we prepared a few examples projects in
+different programming languages and frameworks. You can find them in our custom
+images repository at:
+
+    https://github.com/scrapinghub/custom-images-examples
+
+The 'shub bootstrap' command clones an example project to the current directory
+so that you can start hacking right away.
+
+Run
+
+    shub bootstrap -l
+
+to get a list of all available example projects, then run
+
+    shub bootstrap PROJECTNAME
+
+to clone it.
+"""
+
+SHORT_HELP = "Clone custom image example project"
+
+
+def list_projects_callback(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    projects = get_available_projects()
+    list_projects(projects)
+    ctx.exit()
+
+
+@click.command(help=HELP, short_help=SHORT_HELP)
+@click.option('-l', '--list', 'list_projects', help='list available projects',
+              is_flag=True, callback=list_projects_callback,
+              expose_value=False, is_eager=True)
+@click.argument('project')
+@click.argument('target_dir', required=False)
+def cli(project, target_dir):
+    target_dir = os.path.normpath(
+        os.path.join(os.getcwd(), target_dir or project))
+    if os.path.exists(target_dir):
+        raise BadParameterException(
+            "Target directory %s already exists, please delete it or supply a "
+            "non-existing target." % target_dir)
+    projects = get_available_projects()
+    if project not in projects:
+        raise NotFoundException(
+            "There is no example project named '%s'. Run 'shub bootstrap -l' "
+            "to get a list of all available projects." % project)
+    click.echo("Downloading custom image examples")
+    repo_zip = get_repo_zip(EXAMPLE_REPO)
+    click.echo("Cloning project '%s' into %s" % (project, target_dir))
+    unzip_project(repo_zip, project=projects[project], target_dir=target_dir)
+
+
+def get_available_projects():
+    try:
+        resp = requests.get(AVAILABLE_PROJECTS_URL)
+        resp.raise_for_status()
+    except (requests.HTTPError, requests.ConnectionError) as e:
+        raise RemoteErrorException(
+            "There was an error while getting the list of available projects "
+            "from GitHub: %s.\n\nPlease check your connection or go to\n  %s\n"
+            "to browse the custom image examples manually."
+            "" % (e, "https://github.com/%s" % EXAMPLE_REPO))
+    return yaml.safe_load(resp.text)
+
+
+# XXX: BEGIN Remove this, used for local debugging
+def _get_available_projects():
+    import os.path
+    projects_path = os.path.join(
+        os.path.dirname(__file__), 'bootstrap_projects.yml')
+    with open(projects_path, 'r') as f:
+        return yaml.safe_load(f)
+
+
+if not os.getenv('USING_TOX'):
+    get_available_projects = _get_available_projects  # noqa
+# XXX: END Remove this
```

Comment:
https://github.com/scrapinghub/custom-images-examples/pull/10 is merged, please cleanup the PR @jdemaeyer